modules/videoheroesBidAdapter.js

```diff
@@ -1,261 +1,286 @@
+
+    if (videoObj.playerSize) {
+      const size = Array.isArray(videoObj.playerSize[0]) ? videoObj.playerSize[0] : videoObj.playerSize;
+      videoObj.w = size[0];
+      videoObj.h = size[1];
+    } else {
+      videoObj.w = 640;
+      videoObj.h = 480;
     }
+
+    return videoObj;
+  }
+}
+
+function getFloor(adUnit, mediaType) {
+  let floor = DEF_FLOOR;
+
+  if (!isFn(adUnit.getFloor)) {
+    return floor;
+  }
+
+  let floorObj = adUnit.getFloor({
+    currency: DEF_FLOOR,
+    mediaType,
+    size: '*'
   });
 
-  return bid;
+  if (isPlainObject(floorObj) && !isNaN(parseFloat(floorObj.floor))) {
+    floor = parseFloat(floorObj.floor) || floor;
+  }
+
+  return floor;
 }
 
-const createNativeRequest = br => {
-  let impObject = {
-    ver: '1.2',
-    assets: []
+function prepareImpression(adUnit) {
+  let mediaType = Object.keys(adUnit.mediaTypes)[0];
+
+  const impObj = {
+    id: adUnit.bidId,
+    secure: window.location.protocol.indexOf('https') !== -1 ? 1 : 0,
+    bidfloor: getFloor(adUnit, mediaType)
   };
 
-  let keys = Object.keys(br.mediaTypes.native);
-
-  for (let key of keys) {
-    const props = NATIVE_ASSETS[key];
-    if (props) {
-      const asset = {
-        required: br.mediaTypes.native[key].required ? 1 : 0,
-        id: props.id,
-        [props.name]: {}
-      };
-
-      if (props.type) asset[props.name]['type'] = props.type;
-      if (br.mediaTypes.native[key].len) asset[props.name]['len'] = br.mediaTypes.native[key].len;
-      if (br.mediaTypes.native[key].sizes && br.mediaTypes.native[key].sizes[0]) {
-        asset[props.name]['w'] = br.mediaTypes.native[key].sizes[0];
-        asset[props.name]['h'] = br.mediaTypes.native[key].sizes[1];
-      }
+  impObj[mediaType] = getMediaTypeValues[mediaType](adUnit);
+}
 
-      impObject.assets.push(asset);
-    }
+function prepareSite(adUnit, request) {
+  let siteObj = {};
+
+  siteObj.publisher = {
+    id: adUnit.params.placementId.toString()
+  };
+
+  siteObj.domain = parseUrl(request.refererInfo.page || request.refererInfo.topmostLocation).hostname;
+  siteObj.page = request.refererInfo.page || request.refererInfo.topmostLocation;
+
+  if (request.refererInfo.ref) {
+    siteObj.site.ref = request.refererInfo.ref;
   }
 
-  return impObject;
+  return siteObj;
 }
 
-const createBannerRequest = br => {
-  let size = [];
+function prepareDevice() {
+  let deviceObj = {};
 
-  if (br.mediaTypes.banner.sizes && Array.isArray(br.mediaTypes.banner.sizes)) {
-    if (Array.isArray(br.mediaTypes.banner.sizes[0])) { size = br.mediaTypes.banner.sizes[0]; } else { size = br.mediaTypes.banner.sizes; }
-  } else size = [300, 250];
+  [deviceObj.w, deviceObj.h] = [screen.width, screen.height];
+  deviceObj.language = navigator.language;
+  deviceObj.dnt = navigator.doNotTrack === '1' ? 1 : 0
+  deviceObj.ua = navigator.userAgent;
 
-  return { id: br.transactionId, w: size[0], h: size[1] };
-};
+  return deviceObj;
```

Comment:
why not just take the device object core has prepared for you on bidderRequest.ortb2.device? eg https://github.com/prebid/Prebid.js/pull/12053/files

Reply:
never used it before on any prebid connectors, interesting, so if it can replace all device regular data, lets update it